slackbotExercise.py

```diff
@@ -399,17 +478,14 @@ def listenForCommands(bot):
         text = message['text'].lower()
         if text.startswith(command_start):
 
-            # Check for messages specific to a workout
-            for exercise in bot.exercises:
-                found_exercise = False
-                listen_names = exercise['listenNames'].split(';')
-                for listen_name in listen_names:
-                    if listen_name in text:
-                        requests.post(bot.post_message_URL + "&text=" + exercise['tutorial'])
-                        found_exercise = True
-                        break
-                if found_exercise:
-                    break
+            if 'challenge' in text:
+                initiateThrowdown(bot, all_employees, message)
+                break
+
+            exercise = findExerciseInText(bot, text)
+            if exercise != False:
```

Comment:
https://github.com/hudl/slackbot-workout/pull/12/files#diff-b056e345972afdfd3b970ecf8e6685a9R403

@gregdickhudl I return False if I don't find an exercise name in the text. This might not be very intuitive.